services/fluxmonitor_test.go

```diff
@@ -164,62 +168,233 @@ func TestNewFluxMonitor(t *testing.T) {
 			0,
 			"50500",
 		},
+		{
+			"3 adapters, no threshold",
+			[]string{"50500", "51000", "50000"},
+			0,
+			0,
+			15 * time.Millisecond,
+			3 * time.Millisecond,
+			"50500",
+		},
+		{
+			"3 adapters, 1 non-expected response",
+			[]string{"50000", "51000", "wrong"},
+			0.01,
+			600,
+			15 * time.Millisecond,
+			3 * time.Millisecond,
+			"50500",
+		},
 	}
 	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			var mockAdapters []url.URL
-			for _, v := range tt.adapterResults {
-				mockAdapters = append(mockAdapters, *createMockAdapter(v))
-			}
+		var mockAdapters []url.URL
+		for _, v := range tt.adapterResults {
+			mockAdapters = append(mockAdapters, *createMockAdapter(v))
+		}
 
-			triggerJobRun := make(chan subscriber.Event, 10)
-			var fmConfig FluxMonitorConfig
+		triggerJobRun := make(chan subscriber.Event, 10)
+		var fmConfig FluxMonitorConfig
 
-			fmConfig.Adapters = mockAdapters
-			fmConfig.RequestData = `{"from":"BTC","to":"USD"}`
-			fmConfig.Multiply = 18
-			fmConfig.Threshold = tt.threshold
-			fmConfig.AbsoluteThreshold = tt.absoluteThreshold
-			fmConfig.Heartbeat = tt.heartbeat
-			fmConfig.PollInterval = tt.heartbeat
-			fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 1 * time.Second, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 1 * time.Second}
+		fmConfig.Adapters = mockAdapters
+		fmConfig.RequestData = `{"from":"BTC","to":"USD"}`
+		// fmConfig.Multiply = tt.multiply
+		fmConfig.Threshold = tt.threshold
+		fmConfig.AbsoluteThreshold = tt.absoluteThreshold
+		fmConfig.Heartbeat = tt.heartbeat
+		fmConfig.PollInterval = tt.heartbeat
+		fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 100 * time.Millisecond, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 10 * time.Millisecond}
 
+		t.Run("1 new round event tests: "+tt.name, func(t *testing.T) {
+			fmt.Printf("Testing %s", t.Name())
 			fm, err := NewFluxMonitor("test", fmConfig, triggerJobRun, &mockBlockchainManager{})
 			require.NoError(t, err)
 			defer fm.Stop()
 			wg := sync.WaitGroup{}
 			wg.Add(1)
 			fmt.Println(prettyPrint(fm.state))
-			fmt.Println("New round event, initiated: ", fm.state.RoundID+1)
+			fmt.Println("New round event, not initiated: ", fm.state.RoundID+1)
```

Comment:
A lot of print statements could be useful for debugging, but I don't think we should keep them in the tests.

It might also be better to use debugging tools inside the IDE to make it easier to debug as well.

Reply:
I see. Removed the state printing, but left the description of what is going to be ran next, to also have a description of what is going on, instead of having it as a comment. What do you think?

Reply:
So my way of thinking around log printing during tests is that things should be checked as a test instead of printing human-readable output. If all tests are passing, all these log lines are never going to be read by anyone. If tests are failing, it's better to have tests that explicitly catch things that are wrong.

So in short, I think we shouldn't log any extra output in the tests. It's better with clear comments, and developers can use breakpoints to debug issues if that's necessary in the future.